python/google/protobuf/descriptor.py

```diff
@@ -832,8 +832,11 @@ def __init__(self, name, package, options=None, serialized_pb=None,
     super(FileDescriptor, self).__init__(options, 'FileOptions')
 
     if pool is None:
-      from google.protobuf import descriptor_pool
-      pool = descriptor_pool.Default()
+      if api_implementation.Type() == 'cpp':
+        pool = _message.default_pool
```

Comment:
That's a good point. I added this originally because there was a similar check later in `__init__` (previously line 850). If there's no chance of the `cpp` implementation type making it to this code path, I can remove those checks and their related actions from `__init__` entirely.